src/algebraic_topology/fundamental_groupoid/simply_connected.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright (c) 2022 Praneeth Kolichala. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Praneeth Kolichala
+-/
+import algebraic_topology.fundamental_groupoid.induced_maps
+import topology.homotopy.contractible
+import category_theory.punit
+import algebraic_topology.fundamental_groupoid.punit
+
+/-!
+# Simply connected spaces
+This file defines simply connected spaces.
+A topological space is simply connected if its fundamental groupoid is equivalent to `unit`.
+
+## Main theorems
+  - `simply_connected_iff_unique_homotopic` - A space is simply connected if and only if it is
+    nonempty and there is a unique path up to homotopy between any two points
+
+  - `simply_conntected_space.of_contractible` - A contractible space is simply connected
+-/
+noncomputable theory
+
+open category_theory
+open continuous_map
+open_locale continuous_map
+
+/-- A simply connected space is one whose fundamental groupoid is equivalent to `discrete unit` -/
+class simply_connected_space (X : Type*) [topological_space X] : Prop :=
+(equiv_unit [] : nonempty (fundamental_groupoid X ≌ discrete unit))
+
+lemma simply_connected_def (X : Type*) [topological_space X] :
+  simply_connected_space X ↔ nonempty (fundamental_groupoid X ≌ discrete unit) :=
+⟨λ h, @simply_connected_space.equiv_unit X _ h, λ h, ⟨h⟩⟩
+
+lemma simply_connected_iff_unique_homotopic (X : Type*) [topological_space X] :
+  simply_connected_space X ↔ (nonempty X) ∧
+  ∀ (x y : X), nonempty (unique (path.homotopic.quotient x y)) :=
+by { rw [simply_connected_def, equiv_punit_iff_unique], refl, }
+
+namespace simply_connected_space
+variables {X : Type*} [topological_space X] [simply_connected_space X]
+
+/-- We do not make this an instance because `unique` contains data -/
```

Comment:
I don't understand this comment. Once you've put it inside a `nonempty`, there's no more data. Even so, `unique` is a class too, with plenty of instances in mathlib.

Reply:
Sorry, the comment was just to explain why it isn't `instance unique_homotopic (x y X) : unique (path.homotopic.quotient x y)`. Is there an alternative wording that would be preferable?

I don't think `instance unique_homotopic (x y : X) : nonempty (...)` makes a lot of sense to me because it's hard for me to imagine when the existence of a `unique` instance itself would be used as an instance, but I'll probably change it because it can't hurt.

Reply:
I still don't quite understand this response. I've removed the potentially confusing comment, but I didn't change `lemma unique_homotopic (x y : X) : nonempty (unique ...)` to `instance unique_homotopic (x y : X) : nonempty (unique ...)`, because we never use it as an instance, and it would require using the fully qualified name.